addons/nuxeo-ai-core/elements/admin-ai-advanced-export.js

```diff
@@ -0,0 +1,170 @@
+      <nuxeo-input id="split" label="[[i18n('admin.ai.split')]]" value="{{splitProp}}"></nuxeo-input>
+      <paper-button on-tap="_recompute">[[i18n('admin.ai.stats.action')]]</paper-button>
+      <paper-button on-tap="_export">[[i18n('admin.ai.export.action')]]</paper-button>
+    </nuxeo-card>
+    <nuxeo-card heading="[[i18n('admin.ai.stats')]]">
+      <div class="row-container">
+        <span class="totalLabel">[[_computeLabel(_stats, 'total', i18n)]]</span>
+        <span class="totalLabel">[[_computeLabel(_stats, 'count', i18n)]]</span>
+      </div>
+      <nuxeo-data-table id="statsResult" items="[[_filter(_stats)]]">
+        <nuxeo-data-table-column name="[[i18n('admin.ai.stats.field')]]" flex="10">
+          <template>
+            [[item.field]]
+          </template>
+        </nuxeo-data-table-column>
+        <nuxeo-data-table-column name="[[i18n('admin.ai.stats.type')]]" flex="10">
+          <template>
+            [[item.type]]
+          </template>
+        </nuxeo-data-table-column>
+        <nuxeo-data-table-column name="[[i18n('admin.ai.stats.value')]]" flex="50">
+          <template>
+            [[_displayVal(item)]]
+          </template>
+        </nuxeo-data-table-column>
+      </nuxeo-data-table>
+    </nuxeo-card>
+    <nuxeo-card heading="[[i18n('admin.ai.export')]]">
+      <div class="row-container">
+        [[_exportText]]
+      </div>
+    </nuxeo-card>
+  `,
+  is: 'nuxeo-admin-ai-advanced-export',
+  behaviors: [Nuxeo.LayoutBehavior],
+  properties: {
+    inProps: {
+      type: String,
+      value: 'dc:title,file:content',
+    },
+    outProps: {
+      type: String,
+      value: 'dc:lastContributor',
+    },
+    splitProp: {
+      type: String,
+      value: '80',
+    },
+    query: {
+      type: String,
+      value: "SELECT * FROM Document WHERE ecm:primaryType = 'File'",
+    },
+    _exportText: {
+      type: String,
+      value: '',
+    },
+    _stats: {
+      type: String,
+    },
+  },
+  _displayVal(val) {
+    if (val.numericValue != null) {
+      return val.numericValue;
+    }
+    if (val.type === 'terms') {
+      let toReturn = '';
+      for (let i = 0, len = val.value.length; i < len; i++) {
+        toReturn += ` ${val.value[i].key} (${val.value[i].docCount})`;
+      }
+      return toReturn;
+    }
+    return JSON.stringify(val.value);
+  },
+  _filter(items) {
+    return items && items.filter((item) => !(item.type === 'total' || item.type === 'count'));
+  },
+  _computeLabel(theStats, fieldName) {
+    const foundItem = theStats && theStats.find((item) => item.type === fieldName);
+    return foundItem && this.i18n(`admin.ai.export.${fieldName}`, foundItem.numericValue);
+  },
+  _recompute() {
+    this.$.aiStats.params = {
+      query: this.query,
+      inputs: this.inProps,
+      outputs: this.outProps,
+    };
+    this.$.aiStats
+      .execute()
+      .then((response) => {
+        if (response === undefined || response.length === 0) {
```

Comment:
Is `null` a valid response or a special case? If not, this would be shorter:
```suggestion
        if (!response || response.length === 0) {
```